### PR TITLE
Darwin: Simplify test setup and teardown

### PR DESCRIPTION
**Pull request by ksperling-apple**
Avoid relying on dummy tests that are ordered first / last.

### CODE REVIEW

File: src/darwin/Framework/CHIPTests/MTRBackwardsCompatTests.m [hunk 1]
```diff
@@ -91,64 +93,27 @@ - (void)onCommissioningComplete:(NSError *)error
...
-    // Per-test teardown, runs after each test.
-    [super tearDown];
-}
-
-- (void)initStack
-{
-    sStackInitRan = YES;
-
-    XCTestExpectation * expectation = [self expectationWithDescription:@"Pairing Complete"];
+    XCTestExpectation * expectation = [[XCTestExpectation alloc] initWithDescription:@"Pairing Complete"];
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIPTests/MTRBackwardsCompatTests.m:102]: Ah, now I recall why I had not done this.  It was precisely this issue of how to allocate and wait for the expectations!

File: src/darwin/Framework/CHIPTests/MTRPairingTests.m [hunk 2]
```diff
@@ -227,23 +157,51 @@ + (void)shutdownStack
...
+                                                                                 attestationDelegate:attestationDelegate
+                                                                                   failSafeExtension:failSafeExtension];
+    dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.pairing", DISPATCH_QUEUE_SERIAL_WITH_AUTORELEASE_POOL);
+
+    [sController setDeviceControllerDelegate:controllerDelegate queue:callbackQueue];
+
+    NSError * error;
+    __auto_type * payload = [MTRSetupPayload setupPayloadWithOnboardingPayload:kOnboardingPayload error:&error];
+    XCTAssertNotNil(payload);
+    XCTAssertNil(error);
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIPTests/MTRPairingTests.m:183]: I know you just moved this, but
```suggestion
    XCTAssertNil(error);
    XCTAssertNotNil(payload);
```
gives better error reporting.

File: src/darwin/Framework/CHIPTests/TestHelpers/MTRTestResetCommissioneeHelper.m [hunk 3]
```diff
@@ -16,11 +16,11 @@
 
 #import "MTRTestResetCommissioneeHelper.h"
 
-void ResetCommissionee(MTRBaseDevice * device, dispatch_queue_t queue, XCTestCase * testcase, uint16_t commandTimeout)
+void ResetCommissionee(MTRBaseDevice * device, dispatch_queue_t queue, XCTestCase * testcaseUnused, uint16_t commandTimeout)
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIPTests/TestHelpers/MTRTestResetCommissioneeHelper.m:19]: Followup to remove the unused argument, please.

**Review by bzbarsky-apple — APPROVED** [review]
Thank you!

File: src/darwin/Framework/CHIPTests/MTRPairingTests.m (hunk 2 above)
Comment (ksperling-apple) [reply src/darwin/Framework/CHIPTests/MTRPairingTests.m:183]: Ah, I'm not used to having `setContinueAfterFailure: NO`

Comment (ksperling-apple) [reply src/darwin/Framework/CHIPTests/MTRPairingTests.m:183]: Will do this as a follow-up.

**Comment by ksperling-apple** [comment]
Fast-tracking Darwin-only change.

**Pull request merged**